modules/assim.batch/R/pda.emulator.R

```diff
@@ -194,13 +321,33 @@ pda.emulator <- function(settings, params.id=NULL, param.names=NULL, prior.id=NU
       
       ## Convert probabilities back to parameter values
       for(i in 1:n.param) {
-        m[,i] <- eval(prior.fn$qprior[prior.ind][[i]], list(p=mcmc.out[[c]][,i]))
+        m[,i] <- eval(prior.fn$qprior[prior.ind][[i]], list(p=mcmc.out[[c]]$mcmc[,i]))
       }
     }
-      colnames(m) <- pname[prior.ind]
-      mcmc.out[[c]] <- m
+    colnames(m) <- pname[prior.ind]
+    mcmc.list.tmp[[c]] <- m
+    
+    jvar.list[[c]] <- sqrt(mcmc.out[[c]]$jump@history[nrow(mcmc.out[[c]]$jump@history),])
   }
-
+  
+  
+  if(length(extension.check)==1 & !run.block){
+    
+    # merge with previous run's mcmc samples
+    mcmc.list <- mapply(rbind, mcmc.list, mcmc.list.tmp, SIMPLIFY=FALSE)
+    settings$assim.batch$iter <- nrow(mcmc.list[[1]])
+    burnin <- ifelse(!is.null(settings$assim.batch$burnin), 
+                     as.numeric(settings$assim.batch$burnin), 
+                     ceiling(min(2000,0.2*settings$assim.batch$iter)))
+    
+  }else{
+    
+    mcmc.list <- mcmc.list.tmp
+    burnin <- ifelse(!is.null(settings$assim.batch$burnin), 
+                     as.numeric(settings$assim.batch$burnin), 
+                     ceiling(min(2000,0.2*settings$assim.batch$iter)))
```

Comment:
Also, Don't Repeat Yourself (DRY). It seems like the `burnin` definition here could be moved outside the `if` statement, since it's identical on both sides (and also happens, in identical form, elsewhere in the code).